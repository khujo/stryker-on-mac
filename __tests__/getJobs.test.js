const Iterator = require('../index.js');
const config = require('./data/config.json');
const tree = require('./data/tree.json');

test('Find first parameter without configuration should find "Airbaganzahl" on "Auto" concept', () => {
  let iterator = new Iterator();
  expect(iterator.getJobs(tree)).toMatchObject([{
    containerPath: [],
    parameterIndex: 0,
    parameterName: "Airbaganzahl",
    value: "4",
    concept: 'Auto'
  },{
    containerPath: [],
    parameterIndex: 0,
    parameterName: "Airbaganzahl",
    value: "6",
    concept: 'Auto'
  },{
    containerPath: [],
    parameterIndex: 0,
    parameterName: "Airbaganzahl",
    value: "9",
    concept: 'Auto'
  }]);
});

test('Find first parameter with configuration should find "Airbaganzahl" on "Auto" concept and restrict values', () => {
  let iterator = new Iterator({parameters: [{
    name: "Airbaganzahl",
    concept: "Auto",
    values: ["4","6"]
  }]});
  expect(iterator.getJobs(tree)).toMatchObject([{
    containerPath: [],
    parameterIndex: 0,
    parameterName: "Airbaganzahl",
    value: "4"
  },{
    containerPath: [],
    parameterIndex: 0,
    parameterName: "Airbaganzahl",
    value: "6"
  }]);
});

test('Find first parameter with value restriction on first parameter that restricts to an empty set should return the next parameter', () => {
  let iterator = new Iterator({parameters: [{
    name: "Airbaganzahl",
    concept: "Auto",
    values: ["10","11"]
  }]});
  expect(() => iterator.getJobs(tree)).toThrowErrorMatchingSnapshot();
  try {
    iterator.getJobs(tree);
  } catch (error) {
    expect(error.parameter).toMatchObject({
      name: "Airbaganzahl",
      concept: "Auto"
    });
    expect(error.valueRestriction).toEqual(["10","11"]);
  }
});

test('First parameter should be terminal', () => {
  let iterator = new Iterator({parameters: [{
    name: "Artikelname",
    concept: "Auto"
  },{
    name: "Airbaganzahl",
    concept: "Auto",
    values: ["10","11"]
  }]});
  tree.rootContainer.parameters[1].terminal = true;
  expect(() => iterator.getJobs(tree)).toThrowErrorMatchingSnapshot();
  try {
    iterator.getJobs(tree);
  } catch (error) {
    expect(error.parameter).toMatchObject({
      name: "Airbaganzahl",
      concept: "Auto"
    });
    expect(error.valueRestriction).toEqual(["10","11"]);
  }
});
