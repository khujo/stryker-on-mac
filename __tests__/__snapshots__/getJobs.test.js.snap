// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find first parameter with value restriction on first parameter that restricts to an empty set should return the next parameter 1`] = `"Error finding jobs for parameter \\"Airbaganzahl\\" on concept \\"Auto\\". Current values \\"4,6,9\\" have no intersection with value restriction \\"10,11\\"!"`;

exports[`First parameter should be terminal 1`] = `"Error finding jobs for parameter \\"Airbaganzahl\\" on concept \\"Auto\\". Current values \\"4,6,9\\" have no intersection with value restriction \\"10,11\\"!"`;
